### PR TITLE
Unify resume via experiment API

### PR DESCRIPTION
**Pull request by zequnyu**
This PR will address https://github.com/rlworkgroup/garage/issues/87#issuecomment-505220260.
- Unified resume and train usage in experiment_wrapper
- Updated command line resume and code example

### CODE REVIEW

File: scripts/garage [hunk 1]
```diff
@@ -14,27 +11,32 @@ def cli():
 
 @cli.command()
 @click.option(
-    '--snapshot_dir',
-    required=True,
-    help='Directory of the pickle file to resume experiment from.')
+    '--resume_from_dir', required=True, help='Path to resume experiment from.')
 @click.option(
-    '--resume_epoch',
+    '--resume_from_epoch',
```
Comment (ryanjulian) [inline scripts/garage]: this command will appear as `garage resume` so i think `--from_epoch` and `--from_dir` are still good arguments names, e.g.

`garage resume --from_dir=/data/foo/bar --from_epoch=last`

File: scripts/garage [hunk 2]
```diff
@@ -14,27 +11,32 @@ def cli():
...
+        log_dir = resume_from_dir
+
+    run_experiment(
+        run_task,
+        log_dir=log_dir,
+        resume_from_dir=resume_from_dir,
+        resume_from_epoch=resume_from_epoch)
+
+
+def run_task(_snapshot_config, _variant_data, from_dir, from_epoch):
```
Comment (ryanjulian) [inline scripts/garage]: hmm perhaps just define this inside `resume()`, unless it's used elsewhere. if it is used elsewhere, make it private.

File: scripts/garage (hunk 1 above)
Comment (krzentner) [reply scripts/garage]: Yeah, I agree that this would be much more ergonomic.

Comment (ryanjulian) [reply scripts/garage]: actually, making from_dir positional is probably a better interface:

`garage resume data/foo/bar --from_epoch=last`

Comment (naeioi) [reply scripts/garage]: I second on your last line. handy and pretty.

**Comment by zequnyu** [comment]
This might need force-push. I don't think we can increase coverage of `experiment_wrapper` under current architecture.

**Comment by ryanjulian** [comment]
Please rebase it and let me know when it's ready for force-push (i.e. the CI passed)

**Comment by zequnyu** [comment]
@ryanjulian This is ready.

**Pull request merged**